tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -111,6 +111,12 @@ public int Serialize(ref byte[] bytes, int offset, in TraceChunkModel traceChunk
             // serialize each span
             for (var i = 0; i < traceChunk.SpanCount; i++)
             {
+                if (maxSize != null && offset - originalOffset >= maxSize)
+                {
+                    // We've already reached the maximum size, give up
+                    return 0;
```

Comment:
If we're going to start using integers as error codes, I would have gone with `-1` given _technically_ 0 would be a valid value to return (although _currently_, in practice, we always add the header, some refactoring could change it)

Reply:
Well I went with 0 precisely because it's a valid value to return (since the write was incomplete, we consider that we didn't write anything)